File: app/src/main/java/com/example/mvvm_artichecture_sample/base/network/baseobserver/WebServiceListener.java
package com.example.mvvm_artichecture_sample.base.network.baseobserver;

import com.example.mvvm_artichecture_sample.base.network.model.APIError;

public interface WebServiceListener {
	
	void onSuccess(Object object, int statusCode, String requestType) throws Exception;
	
	void onError(APIError apiError, int statusCode, String requestType) throws Exception;
	
}

File: app/src/main/java/com/example/mvvm_artichecture_sample/base/network/baseobserver/WebServiceListener.kt
package com.example.mvvm_artichecture_sample.base.network.baseobserver

import com.example.mvvm_artichecture_sample.base.network.model.APIError

interface WebServiceListener {

    @Throws(Exception::class)
    fun onSuccess(`object`: Any, statusCode: Int, requestType: String)

    @Throws(Exception::class)
    fun onError(apiError: APIError, statusCode: Int, requestType: String)

}
